### ISSUE TITLE
slime and (read-line) freeze

### ISSUE THREAD

**Issue by joaotavora** [slime/slime#17]
imported from https://bugs.launchpad.net/slime/+bug/690424

When I run (read-line), type a character or two, and press [tab] in slime emacs locks up (i.e. no keyboard response). I've tried without using slime and this does not occur. Perhaps (read-line) is not playing nicely with slime completion?

This was tested using:
sbcl 1.0.45
emacs 23.1.1
slime-cvs (pulled today)
Ubuntu 10.10 (64 bit)

**Comment by lorenschwartz** [comment]
I am having the same issue. One key and enter is find, but as soon as I hit the second key I get the following in the mini buffer:
"; pipelined request... (swank:simple-completions "12" (quote "COMMON-LISP-USER"))

Emacs then hangs indefinitely and I don't seem to have choice but to Force Quit emacs altogether. 

Emacs 26.1
Slime 2.23
Sbcl 1.1.6.0
Mac OS Mojave 10.14.5

**Comment by luismbo** [comment]
@lorenschwartz I'm guessing you're using some sort of auto-complete extension, like slime-company. That should probably be avoided when `slime-repl-read-mode` is on.

Try this workaround:

```elisp
(defun slime-repl-skip-eval-when-reading (slime-eval &rest args)
  (if (slime-reading-p)
      (user-error "Synchronous Lisp Evaluation suppressed while reading input")
      (apply slime-eval args)))

(advice-add 'slime-eval :around #'slime-repl-skip-eval-when-reading)
```

**Comment by luismbo** [comment]
For future reference, here's what's happening:

1. SWANK's repl thread is blocked on `read`.
2. `slime-simple-completions` is triggered.
3. `slime-eval` (with `slime-current-thread` bound to `:repl-thread`) sends an event and `SWANK:DISPATCH-EVENT` dispatches it to the repl thread -- which is busy.
4. `slime-eval` enters an infinite `accept-process-output` loop waiting for the event response.

**Comment by kengregson** [comment]
For anyone landing here (struggling, for example, to work through Practical Common Lisp examples ;-)

This problem still persists/exists without triggering completion with [TAB]. Emacs/Slime hangs after typing even a single character in response to `(read-line *query-io*)` and other interactive input functions. At least with my naive Emacs/Slime configuration.

Extreme thanks to @luismbo for the explanation and workaround to continue working in the SLIME repl!

This was tested using:
sbcl 2.6.7
emacs 30.2
slime 2.32.0.20260820.60
Nix-darwin (aarch64-darwin)

**Comment by stassats** [comment]
Shouldn't happen anymore.

**Comment by kengregson** [comment]
I'll give Commit b412e5f for issue #932 a shot, when I get a chance (also Workaround 2 there) - sounds nearly identical. Thanks @stassats for the pointer and great work!

**Comment by kengregson** [comment]
Verify issue is resolved with ; SLIME 2.32.git using commit https://github.com/slime/slime/commit/b412e5f7d46139e19485d48571e658f9e7c07f1d
